src/Core/src/Handlers/Button/ButtonHandler.Android.cs

```diff
@@ -124,58 +124,12 @@ void OnSetImageSource(Drawable? obj)
 			PlatformView.Icon = obj;
 		}
 
-		bool NeedsExactMeasure()
-		{
-			if (VirtualView.VerticalLayoutAlignment != Primitives.LayoutAlignment.Fill
-				&& VirtualView.HorizontalLayoutAlignment != Primitives.LayoutAlignment.Fill)
-			{
-				// Layout Alignments of Start, Center, and End will be laying out the TextView at its measured size,
-				// so we won't need another pass with MeasureSpecMode.Exactly
-				return false;
-			}
-
-			if (VirtualView.Width >= 0 && VirtualView.Height >= 0)
-			{
-				// If the Width and Height are both explicit, then we've already done MeasureSpecMode.Exactly in 
-				// both dimensions; no need to do it again
-				return false;
-			}
-
-			// We're going to need a second measurement pass so TextView can properly handle alignments
-			return true;
-		}
-
 		public override void PlatformArrange(Rect frame)
 		{
-			var platformView = this.ToPlatform();
-
-			if (platformView == null || Context == null)
-			{
-				return;
-			}
-
-			if (frame.Width < 0 || frame.Height < 0)
-			{
-				return;
-			}
-
-			// Depending on our layout situation, the TextView may need an additional measurement pass at the final size
-			// in order to properly handle any TextAlignment properties.
-			if (NeedsExactMeasure())
-			{
-				platformView.Measure(MakeMeasureSpecExact(frame.Width), MakeMeasureSpecExact(frame.Height));
-			}
-
+			this.PrepareForTextViewArrange(frame);
```

Comment:
Previously, the could would exit out if something was null or -1 and base wouldnot run, but now because that code is extracted, base always runs.

Is this going to cause any issues?

Reply:
The base method also does those checks. And the NeedsExactMeasure logic shouldn't really have been making the "don't run base _at all_ decision" anyway, it should have just been answering the question "do we need an additional exact measure?".